src/Editor/CodelensProvider.ts

```diff
@@ -17,12 +17,14 @@
 import * as vscode from 'vscode';
 
 export class CodelensProvider implements vscode.CodeLensProvider {
+  codeLenses: vscode.CodeLens[] = [];
   constructor() {}
 
   public provideCodeLenses(document: vscode.TextDocument, token: vscode.CancellationToken):
       vscode.CodeLens[]|Thenable<vscode.CodeLens[]> {
-    // TODO implement
-    return [];
+    this.codeLenses = [];
```

Comment:
if `codeLenses` is initialzied in declaration, maybe this line isn't needed... can you please check?

Reply:
As far as I understand, the `activation` function runs exactly once. Therefore, the codelens is also initialized once.
But we have to find the codelens range again if the editor changes.
(If the code changes, only the content cannot be detected. If we can detect the change and know how many times the line is, I think we can reset it just once.)

Reply:
If removing this line is uncertain, let's go as-is for explictly initializing here is safe :)

Reply:
I think your opinion is better because it is only used in the `provideCodeLenses` function.